src/main/java/io/strimzi/kafka/quotas/VolumeSource.java

```diff
@@ -58,62 +64,82 @@ public VolumeSource(Admin admin, VolumeObserver volumeObserver, int timeout, Tim
 
     @Override
     public void run() {
-        log.info("Updating cluster volume usage.");
-        CompletableFuture<Map<Integer, Map<String, LogDirDescription>>> logDirsPerBrokerPromise = new CompletableFuture<>();
-        log.debug("Attempting to describe cluster");
-        admin.describeCluster().nodes().whenComplete((nodes, throwable) -> {
-            if (throwable != null) {
-                log.error("error while describing cluster", throwable);
-                logDirsPerBrokerPromise.completeExceptionally(throwable);
-            } else {
-                if (log.isDebugEnabled()) {
-                    log.debug("Successfully described cluster: " + nodes);
-                }
-                //Deliberately stay on the adminClient thread as the next thing we do is another admin API call
-                onDescribeClusterSuccess(nodes, logDirsPerBrokerPromise);
-            }
-        });
-
         try {
+            log.info("Updating cluster volume usage.");
+            CompletableFuture<VolumeUsageResult> volumeUsagePromise = toResultStage(admin.describeCluster().nodes())
```

Comment:
Yes, we also haven't covered a case mentioned in the proposal [yet](https://github.com/strimzi/proposals/blob/main/047-cluster-wide-volume-usage-quota-management.md#throttle-factor-fallback). If a node drops out of the active set between describeCluster and describeLogDirs calls we could consider this an inconsistency and not act on that data. We'll handle this in another PR.

I'll make an Issue for this problem since it shouldn't block this PR